### ISSUE TITLE
[PROTOCOL] Fix burn function for xERC20

### ISSUE THREAD

**Issue by vmichalik** [superfluid-finance/protocol-monorepo#1865]
## Overview
**What?**
The current implementation of the xERC20 burn function for FRACTION will only work by the Connext Routers themselves

The current Superfluid  `burn` function would actually make a downgrade, something that would be anyways be not the correct behaviour, but also not work for Pure SuperTokens anyways. 
https://github.com/superfluid-finance/protocol-monorepo/issues/1258

A possible approach would be to either fix the current `burn` function so that it works correctly, or change the current xERC20 implementation in a way that their burn function is not exclusive to the predefined Connext Routers anymore.

Personally, I would prefer a function that would allow the Token Deployment account to give a specific account the allowance to burn a specific amount. That would prevent anyone burning tokens randomly (and lowering the totalSupply constantly), but still give the project owners enough flexibilty to do concious burn activites.

Would be awesome if that functionality could be added to the backlog and be bundled with the next Superfluid Upgrade (since doing an Upgrade just for that seems to be quite an overkill).

**How?**

**Why?**

**Deadline:**

**UI/UX Design:**

## Acceptance criteria
 - [ ] ...
 - [ ] Satisfies entire acceptance criteria
 - [ ] Consider communication & visibility
 - [ ] Code review is approved by at least 2 reviewers
 - [ ] Tested by Elvijs/QA & discovered bugs resolved
 - [ ] Documentation updated
 - [ ] Assignee/opened by merges PR
 - [ ] Automated E2E tests for happy path
 - [ ] Code is merged
 - [ ] Observability of critical user path
 - [ ] Steps to access code change as a user

## Links/Resources
 * ...

## Readiness criteria
- [ ] Descriptive title
- [ ] What?
- [ ] Why?
- [ ] How?
- [ ] Acceptance criteria complying with Definition of Done
- [ ] Estimate/Story points
- [ ] Deadline: If not a part of an epic
- [ ] Priority: Reflected as order on the board, walking the board from the top and right
- [ ] Assignee
- [ ] Labels
- [ ] Link to epic: If belonging to an epic
- [ ] A task is always a separate ticket/issue
- [ ] UI/UX Design (optional)
- [ ] Blockers/Prerequisites (optional): Use Zenhub blocked by/blocking feature

**Comment by d10r** [comment]
```
burn(uint256 amount, bytes calldata userData) external);
```
is derived from [ERC-777](https://eips.ethereum.org/EIPS/eip-777) and does indeed behave in a peculiar way in the default SuperToken implementation:
* downgrade if it has an underlying
* revert if it has no underlying

There was an attempt to change behaviour in https://github.com/superfluid-finance/protocol-monorepo/issues/1258, but aborted because there's no obvious right way, so we chose to not change.

The requested possibility to have permissioned burn can already be achieved in Custom Super Tokens. They can use `selfBurn` and add any constraints wanted.

What's probably implied in this requests is to remove the option for token holders to manipulate `totalSupply` by burning.
The way the aforementioned ERC777 burn is implemented, that's already de facto the case for Pure Super Tokens, since it will revert. Also, Custom Super Tokens anyway can change the logic through custom Super Token logic.
